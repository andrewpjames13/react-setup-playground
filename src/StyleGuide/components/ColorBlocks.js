import React, { PureComponent } from 'react';
import styles from './ColorBlocks.scss';

class ColorBlocks extends PureComponent {
  render() {
    return(
      <div className={styles[this.props.color]}>
        <h2>${this.props.color}</h2>
      </div>
    );
  }
}

export default ColorBlocks;
