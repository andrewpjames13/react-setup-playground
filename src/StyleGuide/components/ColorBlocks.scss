@import "~Shared/styles/variables";

@mixin brandColors($backgroundColor, $outlineColor: null, $fontColor: $white) {
  background-color: $backgroundColor;
  outline: 1px solid $outlineColor;
  height: 100px;
  width: 100%;
  color: $fontColor;
}

.primary { @include brandColors($primary); }
.primaryLight { @include brandColors($primaryLight); }
.secondary { @include brandColors($secondary); }
.secondaryMedium { @include brandColors($secondaryMedium); }
.secondaryLight { @include brandColors($secondaryLight); }
