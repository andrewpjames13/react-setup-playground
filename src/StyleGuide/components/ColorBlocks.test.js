import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ColorBlocks from './ColorBlocks';

function setup(color) {
  const props = {
    color
  }

  const colorBlocks = shallow(<ColorBlocks {...props} />)

  return {
    props,
    colorBlocks
  }
}

it('renders the correct class and text', () => {
  const { colorBlocks } = setup('primary');

  expect(colorBlocks.hasClass('primary')).toBe(true)
  expect(colorBlocks.find('h2').text()).toBe('$primary');
});

test('snapshot', () => {
  const { props } = setup('primary');

  const tree = renderer.create(
    <ColorBlocks {...props} />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
