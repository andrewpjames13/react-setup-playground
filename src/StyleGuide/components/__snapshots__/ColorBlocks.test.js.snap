exports[`test snapshot 1`] = `
<div
  className="primary">
  <h2>
    $
    primary
  </h2>
</div>
`;
